system-contracts/test/NonceHolder.spec.ts

```diff
@@ -0,0 +1,185 @@
+import { expect } from "chai";
+import type { NonceHolder } from "../typechain";
+import { NonceHolderFactory } from "../typechain";
+import {
+  TEST_BOOTLOADER_FORMAL_ADDRESS,
+  TEST_DEPLOYER_SYSTEM_CONTRACT_ADDRESS,
+  TEST_NONCE_HOLDER_SYSTEM_CONTRACT_ADDRESS,
+} from "./shared/constants";
+import { prepareEnvironment, setResult } from "./shared/mocks";
+import { deployContractOnAddress, getWallets } from "./shared/utils";
+import { ethers, network } from "hardhat";
+
+describe("NonceHolder tests", () => {
+  const wallet = getWallets()[0];
+  let nonceHolder: NonceHolder;
+  let nonceHolderAccount: ethers.Signer;
+  let deployerAccount: ethers.Signer;
+
+  before(async () => {
+    await prepareEnvironment();
+    await deployContractOnAddress(TEST_NONCE_HOLDER_SYSTEM_CONTRACT_ADDRESS, "NonceHolder");
+    nonceHolder = NonceHolderFactory.connect(TEST_NONCE_HOLDER_SYSTEM_CONTRACT_ADDRESS, wallet);
+    nonceHolderAccount = await ethers.getImpersonatedSigner(TEST_BOOTLOADER_FORMAL_ADDRESS);
```

Comment:
Maybe you can use system caller to make a system calls, instead nonceHolderAccount?

Reply:
Or, at least, add some comment, that system address is needed to make it more clear

Reply:
works fine on the test account, it most likely just a copy-pasting leftover

Reply:
is the current implementation acceptable?

Reply:
It's okay, but I think it makes sense to add a comment, that we are using the system address in order to satisfy system call modifier

Reply:
Yes, but I think it makes sense to add a comment that we are using system address in order to satisfy the system call modifier.

Reply:
commented